base/linalg/lu.jl

```diff
@@ -313,8 +314,9 @@ end
 
 inv!(A::LU{<:BlasFloat,<:StridedMatrix}) =
     @assertnonsingular LAPACK.getri!(A.factors, A.ipiv) A.info
-inv(A::LU{<:BlasFloat,<:StridedMatrix}) =
-    inv!(LU(copy(A.factors), copy(A.ipiv), copy(A.info)))
+inv!(A::LU{T,<:StridedMatrix}) where {T} =
+    @assertnonsingular A_ldiv_B!(A.factors, copy(A), eye(T, size(A,1))) A.info
+inv(A::LU{<:Any,<:StridedMatrix}) = inv!(copy(A))
```

Comment:
I guess we can not use the `inv(A) = inv!(copy(A))` pattern for the non BlasFloat case then. Instead
```julia
inv(A::LU) = A_ldiv_B!(A, eye(size(A)))
```
then. But we do still need to make the copy in the fallback `inv!` such that we can overwrite the stored matrix and use it as the left hand side in the solve at the same time.